### PR TITLE
fix blocking request

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Current endpoint is getting blocked by the following code and only allows single request to serve at a time. 
```py
data = await asyncio.get_running_loop().create_task(get_from_pipe())
```

This PR makes the endpoint concurrent by sending the blocking code to thread.

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by lantiga** [comment]
Good catch @aniketmaurya
We need a test that checking correctness for concurrent requests.

File: tests/test_simple.py [hunk 1]
```diff
@@ -109,7 +108,6 @@ def make_request(port, res_queue):
     res_queue.put(response.json())
 
 
-@pytest.mark.xfail()  # fixme
```
Comment (aniketmaurya) [inline tests/test_simple.py:112]: removed the `xfail` so we know whenever we lose concurrency @lantiga

Comment (aniketmaurya) [reply tests/test_simple.py:112]: seems like I need to change the test function. Looking into it..

File: tests/test_simple.py [hunk 2]
```diff
@@ -121,13 +120,10 @@ def test_concurrent_requests():
...
     res_queue = Queue()
-    req_procs = [Process(target=make_request, args=(port, res_queue), daemon=True) for _ in range(n_requests)]
-
-    for el in req_procs:
-        el.start()
-
-    for el in req_procs:
-        el.join()
+    futures = []
+    with ThreadPoolExecutor(max_workers=n_requests) as executor:
```
Comment (aniketmaurya) [inline tests/test_simple.py:124]: creating 100 processes for making requests would be overkill.

File: tests/test_simple.py (hunk 1 above)
Comment (aniketmaurya) [reply tests/test_simple.py:112]: seems like macos runners are very slow compared to Ubuntu and Windows

**Comment by aniketmaurya** [comment]
@lantiga fixed the tests

**Review by lantiga — APPROVED** [review]
Looks great!

**Pull request merged**